CMakeLists.txt

```diff
@@ -175,16 +175,58 @@ endif(USE_NVTX)
 # - link libraries --------------------------------------------------------------------------------
 
 target_link_libraries(rmm PUBLIC
-  ${CUDART_LIBRARY} cuda spdlog::spdlog_header_only)
+  ${CUDART_LIBRARY} cuda)
+
+# What we really want to do is to add spdlog::spdlog_header_only as a PUBLIC
+# dependency. However, since we don't want to install our own copy of spdlog as part
+# of an RMM install, we can do so only during the build, but not during install, which
+# is what the below achieves.
+# This does, however, mean that downstream use of RMM in a cmake project does not automatically
+# pull in spdlog, ie., if the downstream project wants to use the `logging_resource_adaptor`,
+# it needs to make spdlog available itself.
+target_link_libraries(rmm PRIVATE
+  spdlog::spdlog_header_only)
+set_property(TARGET rmm APPEND PROPERTY
+  INTERFACE_LINK_LIBRARIES $<BUILD_INTERFACE:spdlog::spdlog_header_only>)
 
 ###################################################################################################
 # - install targets -------------------------------------------------------------------------------
 
+include(GNUInstallDirs)
+set(INSTALL_CONFIGDIR ${CMAKE_INSTALL_LIBDIR}/cmake/rmm)
+
+install(EXPORT rmm-targets
+  FILE rmm-targets.cmake
+  NAMESPACE rapids::
+  DESTINATION ${INSTALL_CONFIGDIR}
+  )
+
+include(CMakePackageConfigHelpers)
+configure_package_config_file(
+  rmm-config.cmake.in
+  ${CMAKE_CURRENT_BINARY_DIR}/rmm-config.cmake
+  INSTALL_DESTINATION ${INSTALL_CONFIGDIR}
+  )
+
+write_basic_package_version_file(
+  ${CMAKE_CURRENT_BINARY_DIR}/rmm-config-version.cmake
+  COMPATIBILITY AnyNewerVersion
```

Comment:
We currently make no guarantees about ABI stability except within minor versions, i.e. 0.15.1 will guarantee to be compatible with 0.15.0, but 0.16.0 doesn't make any guarantees with regards to 0.15.0. Should we set this to `SameMinorVersion` instead?

Reply:
Yes, that sounds like the right thing to do, will do.